src/withStyles.jsx

```diff
@@ -51,28 +51,47 @@ export function withStyles(
   /** Cache for storing the result of stylesFn(theme) for all themes. */
   const stylesFnResultCacheMap = typeof WeakMap === 'undefined' ? new Map() : new WeakMap();
 
+  /** Cache for storing the results of computations, theme -> WithStyles -> direction -> results */
+  const withStylesCache = new Map();
+
+  function getComponentCache(theme, component) {
+    const themeCache = withStylesCache.get(theme);
+    if (!themeCache) {
+      return null;
+    }
+    const componentCache = themeCache.get(component);
+    if (!componentCache) {
+      return null;
+    }
+    return componentCache;
+  }
+
+  function getOrCreateComponentCache(theme, component) {
+    let themeCache = withStylesCache.get(theme);
+    if (!themeCache) {
+      themeCache = new Map();
+      withStylesCache.set(theme, themeCache);
+    }
+    let componentCache = themeCache.get(component);
+    if (!componentCache) {
+      componentCache = { rtl: {}, ltr: {} };
+      themeCache.set(component, componentCache);
+    }
+    return componentCache;
+  }
+
+  function updateComponentCache(theme, component, direction, results) {
+    const componentCache = getOrCreateComponentCache(theme, component);
```

Comment:
This the only caller of getOrCreateComponentCache. Put that logic inside updateContentCache.